app/controllers/publish_interface/providers_controller.rb

```diff
@@ -1,6 +1,6 @@
 module PublishInterface
```

Comment:
Might be outside of this PR - but what's the rationale for this module naming? Given that this repo will essentially be the interface for publish as a whole. Would be good if we could stick to module naming based on areas of the app eg `Admin`, `Providers`, `Organisations`, `Support` etc

Reply:
This borrows an idea from Apply. To keep the monolith manageable, Apply adopted an [approach](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/adr/0005-application-structure.md) of separating distinct parts of the app into namespaces (CandidateInterface, ProviderInterface, SupportInterface, VendorAPI, etc). The idea was to establish clear boundaries and keep things loosely coupled between them so the codebase stays manageable/easy to understand over time.

To date, the project's done a pretty decent job of conforming to the structure outlined in that ADR, roughly along the lines of:

- Each namespace
  - Views, view components, controllers, form objects, shared services, mailers
- App layer
  - Models, shared services, shared view components

There's also a lightweight [spec](https://github.com/DFE-Digital/apply-for-teacher-training/blob/6ca59790422975a3c3b97c12276e57e1b9b59d5d/spec/lib/namespaces_spec.rb) for checking we aren't using namespaced code across boundaries.

This may make sense for this project, as its future state is likely to be along the lines of:

- PublishInterface
- SupportInterface
- API
- FindInterface (tbc)

Reply:
Might be worth having a little dev meeting to discuss the approach? Conscious that different teams in DFE use different approaches & methodologies. Given that some of the publish devs are from different teams it's probably worth deciding on what we want to use for this project going forward (and possibly drawing up new ADRs to explain why we went with said approach)